### PR TITLE
Add PragmaSimulationRepetitions

### PR DESCRIPTION
**Pull request by rreiner-hqs**
Add new PRAGMA operation that sets the number of repetitions for the simulation of the whole circuit.

### CODE REVIEW

File: roqoqo/tests/integration/operations/supported_version.rs [hunk 1]
```diff
@@ -391,3 +391,31 @@ fn test_version_1_16_0_three_qubit_gate(operation: operations::ThreeQubitGateOpe
...
+    operations::FourQubitGateOperation::from(operations::TripleControlledPauliX::new(0, 1, 2, 3)); "TripleControlledPauliX"
+)]
+#[test_case(
+    operations::FourQubitGateOperation::from(operations::TripleControlledPauliZ::new(0, 1, 2, 3)); "TripleControlledPauliZ"
+)]
+#[test_case(
+    operations::FourQubitGateOperation::from(operations::TripleControlledPhaseShift::new(0, 1, 2, 3, CalculatorFloat::PI)); "TripleControlledPhaseShift"
+)]
+fn test_version_1_17_0_four_qubit_gate(operation: operations::FourQubitGateOperation) {
+    assert_eq!(operation.minimum_supported_roqoqo_version(), (1, 17, 0));
```
Comment (dberthault) [inline roqoqo/tests/integration/operations/supported_version.rs:420]: Same

File: roqoqo/tests/integration/operations/supported_version.rs [hunk 2]
```diff
@@ -391,3 +391,31 @@ fn test_version_1_16_0_three_qubit_gate(operation: operations::ThreeQubitGateOpe
...
+}
```
Comment (dberthault) [inline roqoqo/tests/integration/operations/supported_version.rs:421]: Missing a test for the minimum supported version of `PragmaSimulationRepetitions`

**Review by dberthault — CHANGES_REQUESTED** [review]
Also don't forget to run cargo build in qoqo to update the pyi files with the new operation

File: roqoqo/tests/integration/operations/supported_version.rs (hunk 2 above)
Comment (rreiner-hqs) [reply roqoqo/tests/integration/operations/supported_version.rs:421]: I added a test for pragmas in 1.17.0

File: roqoqo/tests/integration/operations/supported_version.rs (hunk 1 above)
Comment (rreiner-hqs) [reply roqoqo/tests/integration/operations/supported_version.rs:420]: II've removed this test as there are no new non-pragma operations

File: roqoqo/tests/integration/operations/supported_version.rs [hunk 3]
```diff
@@ -391,3 +391,31 @@ fn test_version_1_16_0_three_qubit_gate(operation: operations::ThreeQubitGateOpe
...
+#[test_case(operations::SingleQubitGateOperation::from(operations::InvSXGate::new(0)); "InvSXGate")]
+fn test_version_1_17_0_single_qubit_gate(operation: operations::SingleQubitGateOperation) {
+    assert_eq!(operation.minimum_supported_roqoqo_version(), (1, 17, 0));
+}
+
+#[test_case(operations::ThreeQubitGateOperation::from(operations::ControlledSWAP::new(0, 1, 2)); "ControlledSWAP")]
+#[test_case(operations::ThreeQubitGateOperation::from(operations::PhaseShiftedControlledControlledZ::new(0, 1, 2, CalculatorFloat::PI)); "PhaseShiftedControlledControlledZ")]
+#[test_case(operations::ThreeQubitGateOperation::from(operations::PhaseShiftedControlledControlledPhase::new(0, 1, 2, CalculatorFloat::PI, CalculatorFloat::PI)); "PhaseShiftedControlledControlledPhase")]
+fn test_version_1_17_0_three_qubit_gate(operation: operations::ThreeQubitGateOperation) {
+    assert_eq!(operation.minimum_supported_roqoqo_version(), (1, 17, 0));
```
[the remark this answers is not in the document]
Comment (rreiner-hqs) [reply roqoqo/tests/integration/operations/supported_version.rs:407]: II've removed this test as there are no new non-pragma operations

File: roqoqo/tests/integration/operations/supported_version.rs [hunk 4]
```diff
@@ -391,3 +391,31 @@ fn test_version_1_16_0_three_qubit_gate(operation: operations::ThreeQubitGateOpe
...
     assert_eq!(operation.minimum_supported_roqoqo_version(), (1, 16, 0));
 }
+
+#[test_case(operations::SingleQubitGateOperation::from(operations::InvTGate::new(0)); "SqrtPaInvTGateuliY")]
+#[test_case(operations::SingleQubitGateOperation::from(operations::InvSGate::new(0)); "InvSGate")]
+#[test_case(operations::SingleQubitGateOperation::from(operations::SXGate::new(0)); "SXGate")]
+#[test_case(operations::SingleQubitGateOperation::from(operations::InvSXGate::new(0)); "InvSXGate")]
+fn test_version_1_17_0_single_qubit_gate(operation: operations::SingleQubitGateOperation) {
+    assert_eq!(operation.minimum_supported_roqoqo_version(), (1, 17, 0));
+}
```
[the remark this answers is not in the document]
Comment (rreiner-hqs) [reply roqoqo/tests/integration/operations/supported_version.rs:401]: II've removed this test as there are no new non-pragma operations

**Review by dberthault — APPROVED** [review]
